### PR TITLE
Remove unused consts from oid lists

### PR DESCRIPTION
**Pull request by palaviv**
Currently when you compile mbedtls with any code that need to use oid's you get a long list of consts that not all of them can be used. For example you will be able to resolve MD2 oid when MD2 operation is not enabled. This PR remove the unused entries from this lists using the config.h defines. This should reduce the size of the executable in most scenarios that use oid's.

### CODE REVIEW

**Comment by sbutcher-arm** [comment]
Thanks for the contribution! This looks like a nice straight forward optimisation for code size.

Looking at the code though, you've indented the preprocessor directives which isn't part of the project's coding style, although this isn't stated explicitly in the [mbed TLS coding standards](https://tls.mbed.org/kb/development/mbedtls-coding-standards), there are no indented directives anywhere else in the source code, and especially not in any comparable code.

Would you mind reformatting the code and resubmitting the PR?

Thanks

**Comment by sbutcher-arm** [comment]
So to accept this or any future contributions we will need a Contributor’s Licence Agreement (CLA) signed or authorised by yourself. You can [find an agreement to sign here](https://www.mbed.com/en/about-mbed/contributor-license-agreements/), which can be signed and returned to us, or you could create an mbed account and [accept a slightly different agreement here with a click through](https://developer.mbed.org/contributor_agreement/) if this is a personal contribution. 

Thanks for your understanding!

**Comment by palaviv** [comment]
I changed the indentation as asked and signed the CLA through your site.

**Comment by sbutcher-arm** [comment]
Thanks for accepting the CLA! Could you please confirm you signed up as user '[Palaviv](https://developer.mbed.org/users/Palaviv/)'?

Thanks

**Comment by palaviv** [comment]
Yes I have.

File: tests/suites/test_suite_debug.data [hunk 1]
```diff
@@ -38,11 +38,11 @@ Debug print buffer #5
...
 
 Debug print certificate #2 (EC)
-depends_on:MBEDTLS_PEM_PARSE_C:MBEDTLS_BASE64_C:MBEDTLS_ECP_C:MBEDTLS_ECP_DP_SECP384R1_ENABLED
+depends_on:MBEDTLS_PEM_PARSE_C:MBEDTLS_BASE64_C:MBEDTLS_ECP_C:MBEDTLS_ECP_DP_SECP384R1_ENABLEDM:BEDTLS_SHA256_C
```
Comment (sbutcher-arm) [inline tests/suites/test_suite_debug.data]: Typo here - should be `MBEDTLS_SHA256_C`

**Comment by sbutcher-arm** [comment]
Reviewed, tested, fixed a minor typo and merged.

**Pull request closed without merging**